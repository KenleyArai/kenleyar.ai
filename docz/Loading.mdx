---
name: Loading
---

import { Playground, PropsTable } from 'docz';
import Loading from 'components/Loading';

# Loading

## Signature

<PropsTable of={Loading} />

## Examples

## While loading state

```
<Loading />
```

<Playground>
  <Loading />
</Playground>

## Failed to load state

```
<Loading failed={true} data={'Error while loading'} />
```

<Playground>
  <Loading failed={true} data={'Error while loading'} />
</Playground>

## Sucessfully loaded

`data` is ignored when the component sucessfully loads

```
<Loading failed={false} data={'Loaded!'} />
```

<Playground>
  <Loading failed={false} data={'Loaded!'} />
</Playground>
