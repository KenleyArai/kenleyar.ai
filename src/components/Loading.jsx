import React from 'react';
import DoneLoader from 'components/styled/DoneLoader';
import StillLoader from 'components/styled/StillLoader';
import FailedLoader from 'components/styled/FailedLoader';
import PropTypes from 'prop-types';

const Loading = ({ failed, data }) => {
  if (failed === undefined || failed === null) {
    return <StillLoader />;
  } else if (failed) {
    return <FailedLoader data={data} />;
  } else {
    return <DoneLoader />;
  }
};

Loading.propTypes = {
  failed: PropTypes.bool,
  data: PropTypes.string,
};

export default Loading;
