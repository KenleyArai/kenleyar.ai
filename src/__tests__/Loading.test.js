// Loading.test.js
import React from 'react';
import Loading from 'components/Loading';
import renderer from 'react-test-renderer';

test('Loading', () => {
  const loading = renderer.create(<Loading />).toJSON();
  expect(loading).toMatchSnapshot();

  const done_loading = renderer.create(<Loading data={'some data'} />).toJSON();
  expect(done_loading).toMatchSnapshot();

  const failed_loader = renderer
    .create(<Loading failed={true} data={'Failed to load'} />)
    .toJSON();
  expect(failed_loader).toMatchSnapshot();
});
