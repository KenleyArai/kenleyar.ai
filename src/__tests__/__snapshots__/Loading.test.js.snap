// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading 1`] = `
Array [
  <img
    src="placeholder.png"
    style={
      Object {
        "opacity": "0.8414709848078965",
        "transform": "translate3d(0, 0px, 0)",
      }
    }
  />,
  <img
    src="placeholder.png"
    style={
      Object {
        "opacity": "0.8414709848078964",
        "transform": "translate3d(0, 12px, 0)",
      }
    }
  />,
  <img
    src="placeholder.png"
    style={
      Object {
        "opacity": "0.8414709848078963",
        "transform": "translate3d(0, 7.34788079488412e-15px, 0)",
      }
    }
  />,
  <img
    src="placeholder.png"
    style={
      Object {
        "opacity": "0.8414709848078962",
        "transform": "translate3d(0, -12px, 0)",
      }
    }
  />,
  <img
    src="placeholder.png"
    style={
      Object {
        "opacity": "0.841470984807896",
        "transform": "translate3d(0, -1.469576158976824e-14px, 0)",
      }
    }
  />,
]
`;

exports[`Loading 2`] = `
Array [
  <img
    src="placeholder.png"
    style={
      Object {
        "opacity": "0.8414709848078965",
        "transform": "translate3d(0, 0px, 0)",
      }
    }
  />,
  <img
    src="placeholder.png"
    style={
      Object {
        "opacity": "0.8414709848078964",
        "transform": "translate3d(0, 12px, 0)",
      }
    }
  />,
  <img
    src="placeholder.png"
    style={
      Object {
        "opacity": "0.8414709848078963",
        "transform": "translate3d(0, 7.34788079488412e-15px, 0)",
      }
    }
  />,
  <img
    src="placeholder.png"
    style={
      Object {
        "opacity": "0.8414709848078962",
        "transform": "translate3d(0, -12px, 0)",
      }
    }
  />,
  <img
    src="placeholder.png"
    style={
      Object {
        "opacity": "0.841470984807896",
        "transform": "translate3d(0, -1.469576158976824e-14px, 0)",
      }
    }
  />,
]
`;

exports[`Loading 3`] = `"Failed to load"`;
